substratevm/src/com.oracle.graal.pointsto/src/com/oracle/graal/pointsto/heap/ImageHeapScanner.java

```diff
@@ -464,6 +464,9 @@ protected String formatReason(String message, ScanReason reason) {
     protected ValueSupplier<JavaConstant> readHostedFieldValue(AnalysisField field, JavaConstant receiver) {
         // Wrap the hosted constant into a substrate constant
         JavaConstant value = universe.lookup(hostedConstantReflection.readFieldValue(field.wrapped, receiver));
+        if (value == null) {
+            value = JavaConstant.NULL_POINTER;
```

Comment:
`null` and `JavaConstant.NULL_POINTER` are not semantically equivalent in this case. If `hostedConstantReflection.readFieldValue()` returns `null` then something is potentially wrong and the caller of this needs to be defensive. The hosted provider in this case is [`HotSpotConstantReflectionProvider`](https://github.com/openjdk/jdk/blob/master/src/jdk.internal.vm.ci/share/classes/jdk.vm.ci.hotspot/src/jdk/vm/ci/hotspot/HotSpotConstantReflectionProvider.java#L164).

Reply:
As there is no class initialization at standalone pointsto analysis time, there is no value for static fields. This assignment was to avoid the NPE. You're right, it confuses the `null` and `NULL_POINTER` sematic.
I added `StandaloneImageHeapScanner#isValueAvailable` to override the `ImageHeapScanner#isValueAvailable` to mark the static field's value is not available, so that the value shall never be read.